lib/ecto/association.ex

```diff
@@ -382,6 +383,38 @@ defmodule Ecto.Association do
               "expected defaults for #{inspect name} to be a keyword list " <>
                 "or a {module, fun, args} tuple, got: `#{inspect defaults}`"
 
+  @doc """
+  Validates `preload_order` for association named `name`.
+  """
+  def validate_preload_order!(name, preload_order) when is_list(preload_order) do
+    Enum.map(preload_order, fn
+      field when is_atom(field) ->
```

Comment:
I think it would be great if we managed to validate these fields on `__after_compile__`, if possible, but I'm not sure if it is.

Reply:
I think would be nice to validate the fields. But maybe we should introduce to other places as well, like the `related_key` option. WDYT @v0idpwn ?

@josevalim There is any plans to introduce field validation at compile time?